src/client/dfs/SConscript

```diff
@@ -20,10 +23,12 @@ def configure_lustre(denv):
             gotversion = True
     if gotversion is True:
         print("Building with Lustre bindings.")
-        conf.env.AppendUnique(CCFLAGS=['-DLUSTRE_INCLUDE'])
+        # add compile flag to orig env
+        env.AppendUnique(CCFLAGS=['-DLUSTRE_INCLUDE'])
     else:
         print("Not building with Lustre bindings.")
-    return conf.Finish()
+    conf.Finish()
+    return env
```

Comment:
I think you should instead change line 39 from

denv = configure_lustre(denv)

to 

configure_lustre(denv)

If you do this, the only change you need to make is on line 23 (27 in the new code) to change denv rather than conf.env.

Configure already clones the environment internally.

Reply:
> Configure already clones the environment internally.

Ah ok ..., after the autmatic+silent add-on of found lib in LIBS “construction variable” by CheckLibWithHeader(), a new hidden behaviour within Scons I was ignoring :-( 
Well, Better understand now why you found my previous solution a bit too complex ;-) , thx to teach me !